File: 两数之和/twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        int n=nums.size();
        unordered_map<int, int> hashTab;
        for (int i=0; i<n; i++) {
            auto it = hashTab.find(target - nums[i]);
            if (it != hashTab.end()) {
                res.push_back(i);
                res.push_back(it->second);
                break;
            } else {
                hashTab[nums[i]] = i;
            }
        }
        return res;
    }
};

File: 两数之和/twoSum.go
//使用hashTab通过一次遍历找到目标和对应的数组中两个数的下标
//算法时间复杂度O(n)
func twoSum(nums []int, target int) []int {
    var hashTab map[int]int = make(map[int]int)
    for i,n := range nums {
        if p, ok := hashTab[target-n]; ok {
            return []int{i, p}
        } else {
            hashTab[n] = i
        }
    }
    return nil
}
